src/main/java/org/folio/search/service/browse/ClassificationBrowseService.java

```diff
@@ -0,0 +1,136 @@
+  }
+
+  @Override
+  protected SearchSourceBuilder getAnchorSearchQuery(BrowseRequest req, BrowseContext ctx) {
+    log.debug("getAnchorSearchQuery:: by [request: {}]", req);
+    var config = configService.getConfig(BrowseType.INSTANCE_CLASSIFICATION, req.getBrowseOptionType());
+    var termQueryBuilder = getQuery(ctx, config, termQuery(req.getTargetField(), ctx.getAnchor()));
+    var query = consortiumSearchHelper.filterBrowseQueryForActiveAffiliation(ctx,
+      termQueryBuilder, req.getResource());
+    return searchSource().query(query)
+      .size(ctx.getLimit(ctx.isBrowsingForward()))
+      .from(0);
+  }
+
+  @Override
+  protected SearchSourceBuilder getSearchQuery(BrowseRequest req, BrowseContext ctx, boolean isBrowsingForward) {
+    log.debug("getSearchQuery:: by [request: {}, isBrowsingForward: {}]", req, isBrowsingForward);
+    var config = configService.getConfig(BrowseType.INSTANCE_CLASSIFICATION, req.getBrowseOptionType());
+
+    var browseField = getBrowseField(config);
+    var normalizedAnchor = ShelvingOrderCalculationHelper.calculate(ctx.getAnchor(), config.getShelvingAlgorithm());
+    var query = consortiumSearchHelper.filterBrowseQueryForActiveAffiliation(ctx, getQuery(ctx, config, null),
+      req.getResource());
+
+    return searchSource().query(query)
+      .searchAfter(new Object[] {normalizedAnchor.toLowerCase(ROOT)})
+      .sort(fieldSort(browseField).order(isBrowsingForward ? ASC : DESC))
+      .size(ctx.getLimit(isBrowsingForward) + 1)
+      .from(0);
+  }
+
+  @Override
+  protected ClassificationNumberBrowseItem getEmptyBrowseItem(BrowseContext context) {
+    return new ClassificationNumberBrowseItem()
+      .classificationNumber(context.getAnchor())
+      .totalRecords(0)
+      .isAnchor(true);
+  }
+
+  @Override
+  protected BrowseResult<ClassificationNumberBrowseItem> mapToBrowseResult(BrowseContext ctx,
+                                                                           SearchResult<ClassificationResource> res,
+                                                                           boolean isAnchor) {
+    return BrowseResult.of(res)
+      .map(resource -> new ClassificationNumberBrowseItem()
+        .classificationNumber(resource.number())
+        .classificationTypeId(resource.typeId())
+        .isAnchor(isAnchor ? true : null)
+        .totalRecords(getTotalRecords(ctx, resource)));
+  }
+
+  private static QueryBuilder getQuery(BrowseContext ctx, BrowseConfig config, TermQueryBuilder anchorQuery) {
+    var typeIds = config.getTypeIds();
+    if (config.getTypeIds().isEmpty() && ctx.getFilters().isEmpty()) {
+      if (anchorQuery != null) {
+        return anchorQuery;
+      }
+      return matchAllQuery();
+    } else {
+      var boolQueryMain = boolQuery();
+      if (!config.getTypeIds().isEmpty()) {
+        var boolQuery = boolQuery();
+        for (var typeId : typeIds) {
+          boolQuery.should(termQuery(CLASSIFICATION_TYPE_ID_FIELD, typeId.toString()));
```

Comment:
You can add "should" clauses to a main query if you have "minimumShouldMatch(1)" in it. Unless there's any other reason to use such query nesting

Reply:
It's look more clear to me.
And also I think that in consortium there will be added additional active affiliation should queries that could break filtering logic.